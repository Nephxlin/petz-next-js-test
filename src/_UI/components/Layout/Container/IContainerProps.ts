export interface IContainerProps {
  children: React.ReactNode
  stylepattern?: 'primary' | 'white'
}