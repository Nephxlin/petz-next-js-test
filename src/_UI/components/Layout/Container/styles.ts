import styled from "styled-components";
import { IContainerProps } from "./IContainerProps";

const stylePatternProps = {
  primary: {
    backgroundColor: '#E40F0F',
    color: '#FFFFFF',
  }
}

export const ContainerStyled = styled.div<IContainerProps>`
  width: 100%;
  ${({$stylePattern}) => $stylePattern === 'primary' && stylePatternProps.primary};
`

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  margin: 0 auto;
  max-width: 1440px;
  padding: 32px 106px;
  padding-bottom: 60px;
  width: 100%;
`