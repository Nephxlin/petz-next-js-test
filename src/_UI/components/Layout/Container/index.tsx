import * as S from './styles'
import { IContainerProps } from "./IContainerProps"

const Container = ({ children, stylepattern }: IContainerProps) => {
  return (
    <S.ContainerStyled data-testid='container-styled' stylepattern={stylepattern}>
      <S.Container>
        {children}
      </S.Container>
    </S.ContainerStyled>
  )
}

export default Container