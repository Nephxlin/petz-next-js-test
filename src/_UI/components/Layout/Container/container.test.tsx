import React from 'react';
import { render, screen } from '@testing-library/react';
import Container from './index';
import 'jest-styled-components';

describe('Container component', () => {
  test('renders Container component with default stylepattern', () => {
    render(<Container stylepattern='white'>Hello, World!</Container>);
    const container = screen.getByTestId('container-styled');
    expect(container).toHaveTextContent('Hello, World!');

    expect(container).toMatchSnapshot();
  });

  test('renders Container component with custom stylepattern', () => {
    render(<Container stylepattern='primary'>Hello World!</Container>);
    const container = screen.getByTestId('container-styled');
    expect(container).toHaveTextContent('Hello World!');
    expect(container).toHaveStyleRule('background-color', '#E40F0F');
    expect(container).toMatchSnapshot();
  });
});
