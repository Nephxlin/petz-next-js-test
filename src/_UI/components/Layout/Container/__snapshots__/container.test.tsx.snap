// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component renders Container component with custom stylepattern 1`] = `
.c0 {
  width: 100%;
  background-color: #E40F0F;
  color: #FFFFFF;
}

.c1 {
  display: flex;
  flex-direction: column;
  margin: 0 auto;
  max-width: 1440px;
  padding: 32px 106px;
  padding-bottom: 60px;
  width: 100%;
}

<div
  class="c0"
  data-testid="container-styled"
  stylepattern="primary"
>
  <div
    class="c1"
  >
    Hello World!
  </div>
</div>
`;

exports[`Container component renders Container component with default stylepattern 1`] = `
.c0 {
  width: 100%;
}

.c1 {
  display: flex;
  flex-direction: column;
  margin: 0 auto;
  max-width: 1440px;
  padding: 32px 106px;
  padding-bottom: 60px;
  width: 100%;
}

<div
  class="c0"
  data-testid="container-styled"
  stylepattern="white"
>
  <div
    class="c1"
  >
    Hello, World!
  </div>
</div>
`;
